activerecord/lib/active_record/connection_adapters/postgresql/oid/interval.rb

```diff
@@ -0,0 +1,42 @@
+require 'active_support/duration'
+
+module ActiveRecord
+  module ConnectionAdapters
+    module PostgreSQL
+      module OID # :nodoc:
+        class Interval < Type::Value # :nodoc:
+          def type
+            :interval
+          end
+
+          def cast_value(value)
+            case value
+            when ::ActiveSupport::Duration
+              value
+            when ::String
+              # Will raise +ActiveSupport::Duration::ISO8601Parser::ParsingError+ on string of unknown format
+              ::ActiveSupport::Duration.parse(value)
```

Comment:
This method should not raise on an invalid string. It should return `nil` in order to be consistent with the rest of the Active Record types.